lib/chkname.c

```diff
@@ -74,23 +72,28 @@ static bool is_valid_name (const char *name)
 	return !numeric;
 }
 
-bool is_valid_user_name (const char *name)
+
+bool
+is_valid_user_name(const char *name)
 {
-	long  maxlen;
+	long    conf;
+	size_t  maxsize;
 
-	/*
-	 * User name length should be limited by the kernel
-	 */
 	errno = 0;
-	maxlen = sysconf(_SC_LOGIN_NAME_MAX);
-	if (maxlen == -1 && errno != 0)
-		maxlen = LOGIN_NAME_MAX;
-	if (maxlen != -1 && strlen(name) >= (size_t)maxlen)
+	conf = sysconf(_SC_LOGIN_NAME_MAX);
+
+	if (conf == -1 && errno != 0)
+		maxsize = LOGIN_NAME_MAX;
+	else
+		maxsize = conf;
```

Comment:
I'd suggest a `(size_t)` cast here, just to make clear that you are aware of the signed/unsigned difference here and that it's not a mistake.

No technical difference though.